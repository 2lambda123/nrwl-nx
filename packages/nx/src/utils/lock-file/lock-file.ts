import { readFileSync, writeFileSync } from 'fs-extra';
import { detectPackageManager, PackageManager } from '../package-manager';
import { parseYarnLockFile, stringifyYarnLockFile } from './yarn';
import { parseNpmLockFile, stringifyNpmLockFile } from './npm';
import { parsePnpmLockFile, stringifyPnpmLockFile } from './pnpm';
import { LockFileData } from './lock-file-type';

export function parseLockFile(
  packageManager: PackageManager = detectPackageManager()
): LockFileData {
  if (packageManager === 'yarn') {
    const file = readFileSync('yarn.lock', 'utf8');
    return parseYarnLockFile(file);
  }
  if (packageManager === 'pnpm') {
    const file = readFileSync('pnpm-lock.yaml', 'utf8');
    return parsePnpmLockFile(file);
  }
  if (packageManager === 'npm') {
    const file = readFileSync('package-lock.json', 'utf8');
    return parseNpmLockFile(file);
  }
  throw Error(`Unknown package manager: ${packageManager}`);
}

export function writeLockFile(
  lockFile: LockFileData,
  packageManager: PackageManager = detectPackageManager()
): void {
  if (packageManager === 'yarn') {
    const content = stringifyYarnLockFile(lockFile);
    writeFileSync('yarn.lock', content);
    return;
  }
  if (packageManager === 'pnpm') {
    const content = stringifyPnpmLockFile(lockFile);
    writeFileSync('pnpm-lock.yaml', content);
    return;
  }
  if (packageManager === 'npm') {
    const content = stringifyNpmLockFile(lockFile);
    writeFileSync('package-lock.json', content);
    return;
  }
  throw Error(`Unknown package manager: ${packageManager}`);
}
