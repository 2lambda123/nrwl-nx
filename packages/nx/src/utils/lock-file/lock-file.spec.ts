import { mapLockFileDataToPartialGraph } from './lock-file';
import { parseNpmLockFile } from './npm';
import { parsePnpmLockFile } from './pnpm';
import { parseYarnLockFile } from './yarn';
import {
  lockFileV3YargsAndDevkitOnly as npmLockFileV3YargsAndDevkit,
  lockFileV2YargsAndDevkitOnly as npmLockFileV2YargsAndDevkit,
  lockFileV1YargsAndDevkitOnly as npmLockFileV1YargsAndDevkit,
  lockFileV3 as npmLockFileV3,
  lockFileV2 as npmLockFileV2,
  lockFileV1 as npmLockFileV1,
} from './__fixtures__/npm.lock';
import {
  lockFileYargsOnly as pnpmLockFileYargsOnly,
  lockFile as pnpmLockFile,
} from './__fixtures__/pnpm.lock';
import {
  lockFileDevkitAndYargs as yarnLockFileDevkitAndYargs,
  lockFile as yarnLockFile,
} from './__fixtures__/yarn.lock';
import { vol } from 'memfs';

jest.mock('fs', () => require('memfs').fs);

jest.mock('@nrwl/devkit', () => ({
  ...jest.requireActual<any>('@nrwl/devkit'),
  workspaceRoot: '/root',
}));

jest.mock('nx/src/utils/workspace-root', () => ({
  workspaceRoot: '/root',
}));

describe('lock-file', () => {
  describe('mapLockFileDataToExternalNodes', () => {
    describe('yarn', () => {
      const fileSys = {
        'node_modules/chalk/package.json': '{"version": "4.1.0"}',
        'node_modules/glob/package.json': '{"version": "7.1.4"}',
        'node_modules/js-yaml/package.json': '{"version": "4.1.0"}',
        'node_modules/minimatch/package.json': '{"version": "3.0.5"}',
        'node_modules/semver/package.json': '{"version": "7.3.4"}',
        'node_modules/tslib/package.json': '{"version": "2.4.0"}',
        'node_modules/yargs-parser/package.json': '{"version": "21.0.1"}',
      };
      beforeEach(() => {
        vol.fromJSON(fileSys, '/root');
      });

      it('should map lock file data to external nodes', () => {
        const lockFileData = parseYarnLockFile(yarnLockFileDevkitAndYargs);

        const partialGraph = mapLockFileDataToPartialGraph(
          lockFileData,
          'yarn'
        );

        expect(partialGraph.externalNodes['npm:yargs']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:yargs']).toMatchSnapshot();
      });

      it('should map successfully complex lock file', () => {
        const lockFileData = parseYarnLockFile(yarnLockFile);

        const partialGraph = mapLockFileDataToPartialGraph(
          lockFileData,
          'yarn'
        );

        expect(partialGraph.externalNodes['npm:nx']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:nx']).toMatchSnapshot();
      });
    });

    describe('npm', () => {
      const fileSys = {
        'node_modules/chalk/package.json': '{"version": "4.1.2"}',
        'node_modules/glob/package.json': '{"version": "7.1.4"}',
        'node_modules/js-yaml/package.json': '{"version": "4.1.0"}',
        'node_modules/minimatch/package.json': '{"version": "4.0.5"}',
        'node_modules/semver/package.json': '{"version": "7.3.4"}',
        'node_modules/tslib/package.json': '{"version": "2.4.1"}',
        'node_modules/yargs-parser/package.json': '{"version": "21.1.1"}',
      };
      beforeEach(() => {
        vol.fromJSON(fileSys, '/root');
      });

      it('should map lock file v3 data to external nodes', () => {
        const lockFileData = parseNpmLockFile(npmLockFileV3YargsAndDevkit);

        const partialGraph = mapLockFileDataToPartialGraph(lockFileData, 'npm');

        expect(partialGraph.externalNodes['npm:yargs']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:yargs']).toMatchSnapshot();
      });

      it('should map lock file v2 data to external nodes', () => {
        const lockFileData = parseNpmLockFile(npmLockFileV2YargsAndDevkit);

        const partialGraph = mapLockFileDataToPartialGraph(lockFileData, 'npm');

        expect(partialGraph.externalNodes['npm:yargs']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:yargs']).toMatchSnapshot();
      });

      it('should map lock file v1 data to external nodes', () => {
        const lockFileData = parseNpmLockFile(npmLockFileV1YargsAndDevkit);

        const partialGraph = mapLockFileDataToPartialGraph(lockFileData, 'npm');

        expect(partialGraph.externalNodes['npm:yargs']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:yargs']).toMatchSnapshot();
      });

      it('should map successfully complex lock file v3', () => {
        const lockFileData = parseNpmLockFile(npmLockFileV3);

        const partialGraph = mapLockFileDataToPartialGraph(lockFileData, 'npm');

        expect(partialGraph.externalNodes['npm:nx']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:nx']).toMatchSnapshot();
      });

      it('should map successfully complex lock file v2', () => {
        const lockFileData = parseNpmLockFile(npmLockFileV2);

        const partialGraph = mapLockFileDataToPartialGraph(lockFileData, 'npm');

        expect(partialGraph.externalNodes['npm:nx']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:nx']).toMatchSnapshot();
      });

      it('should map successfully complex npm lock file v1', () => {
        const lockFileData = parseNpmLockFile(npmLockFileV1);

        const partialGraph = mapLockFileDataToPartialGraph(lockFileData, 'npm');

        expect(partialGraph.externalNodes['npm:nx']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:nx']).toMatchSnapshot();
      });
    });

    describe('pnpm', () => {
      const fileSys = {
        'node_modules/chalk/package.json': '{"version": "4.1.0"}',
        'node_modules/glob/package.json': '{"version": "7.1.4"}',
        'node_modules/js-yaml/package.json': '{"version": "4.1.0"}',
        'node_modules/minimatch/package.json': '{"version": "3.0.5"}',
        'node_modules/semver/package.json': '{"version": "7.3.4"}',
        'node_modules/tslib/package.json': '{"version": "2.4.0"}',
        'node_modules/yargs-parser/package.json': '{"version": "21.0.1"}',
      };
      beforeEach(() => {
        vol.fromJSON(fileSys, '/root');
      });

      it('should map lock file data to external nodes', () => {
        const lockFileData = parsePnpmLockFile(pnpmLockFileYargsOnly);

        const partialGraph = mapLockFileDataToPartialGraph(
          lockFileData,
          'pnpm'
        );

        expect(partialGraph.externalNodes['npm:yargs']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:yargs']).toMatchSnapshot();
      });

      it('should map successfully complex lock file', () => {
        const lockFileData = parsePnpmLockFile(pnpmLockFile);

        const partialGraph = mapLockFileDataToPartialGraph(
          lockFileData,
          'pnpm'
        );

        expect(partialGraph.externalNodes['npm:nx']).toMatchSnapshot();
        expect(partialGraph.dependencies['npm:nx']).toMatchSnapshot();
        expect(
          partialGraph.externalNodes['npm:@phenomnomnominal/tsquery']
        ).toMatchSnapshot();
        expect(
          partialGraph.dependencies['npm:@phenomnomnominal/tsquery']
        ).toMatchSnapshot();
      });
    });
  });
});
