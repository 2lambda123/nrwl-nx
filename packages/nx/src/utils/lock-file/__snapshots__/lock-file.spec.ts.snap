// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lock-file mapLockFileDataToExternalNodes npm should map lock file v1 data to external nodes 1`] = `
Object {
  "data": Object {
    "hash": "c22918aa52c9fdce78bf82451d8114e00b6d4375a8e5b1d1d96c6cf4755dbe1a",
    "packageName": "yargs",
    "version": "17.5.1",
  },
  "name": "npm:yargs",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map lock file v1 data to external nodes 2`] = `
Array [
  Object {
    "source": "npm:yargs",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:escalade",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:get-caller-file",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:require-directory",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:y18n",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map lock file v2 data to external nodes 1`] = `
Object {
  "data": Object {
    "hash": "c22918aa52c9fdce78bf82451d8114e00b6d4375a8e5b1d1d96c6cf4755dbe1a",
    "packageName": "yargs",
    "version": "17.5.1",
  },
  "name": "npm:yargs",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map lock file v2 data to external nodes 2`] = `
Array [
  Object {
    "source": "npm:yargs",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:escalade",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:get-caller-file",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:require-directory",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:y18n",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map lock file v3 data to external nodes 1`] = `
Object {
  "data": Object {
    "hash": "c22918aa52c9fdce78bf82451d8114e00b6d4375a8e5b1d1d96c6cf4755dbe1a",
    "packageName": "yargs",
    "version": "17.5.1",
  },
  "name": "npm:yargs",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map lock file v3 data to external nodes 2`] = `
Array [
  Object {
    "source": "npm:yargs",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:escalade",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:get-caller-file",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:require-directory",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:y18n",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map successfully complex lock file v2 1`] = `
Object {
  "data": Object {
    "hash": "8da365ef8d327d868461fda5c863f779c2f75acfe6a591e733b31a2882763af3",
    "packageName": "nx",
    "version": "14.7.5",
  },
  "name": "npm:nx",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map successfully complex lock file v2 2`] = `
Array [
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/cli",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/tao",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@parcel/watcher",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chalk@4.1.0",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chokidar",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-cursor",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-spinners",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:dotenv",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:enquirer",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fast-glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:figures",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:flat",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fs-extra",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:ignore",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:js-yaml",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:jsonc-parser",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:minimatch@3.0.5",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:npm-run-path",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:open",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:semver",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tar-stream",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tmp",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tsconfig-paths",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tslib",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:v8-compile-cache",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs-parser@21.0.1",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map successfully complex lock file v3 1`] = `
Object {
  "data": Object {
    "hash": "8da365ef8d327d868461fda5c863f779c2f75acfe6a591e733b31a2882763af3",
    "packageName": "nx",
    "version": "14.7.5",
  },
  "name": "npm:nx",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map successfully complex lock file v3 2`] = `
Array [
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/cli",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/tao",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@parcel/watcher",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chalk@4.1.0",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chokidar",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-cursor",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-spinners",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:dotenv",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:enquirer",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fast-glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:figures",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:flat",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fs-extra",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:ignore",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:js-yaml",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:jsonc-parser",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:minimatch@3.0.5",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:npm-run-path",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:open",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:semver",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tar-stream",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tmp",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tsconfig-paths",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tslib",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:v8-compile-cache",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs-parser@21.0.1",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map successfully complex npm lock file v1 1`] = `
Object {
  "data": Object {
    "hash": "62016caa192e50e3af6e4730fa1427455ad431a0e11ad807215c56bee04a8c8e",
    "packageName": "nx",
    "version": "14.7.5",
  },
  "name": "npm:nx",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes npm should map successfully complex npm lock file v1 2`] = `
Array [
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/cli",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/tao",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@parcel/watcher",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chalk@4.1.0",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chokidar",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-cursor",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-spinners",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:dotenv",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:enquirer",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fast-glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:figures",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:flat",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fs-extra",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:ignore",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:js-yaml",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:jsonc-parser",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:minimatch@3.0.5",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:npm-run-path",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:open",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:semver",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tar-stream",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tmp",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tsconfig-paths",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tslib",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:v8-compile-cache",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs-parser@21.0.1",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes pnpm should map lock file data to external nodes 1`] = `
Object {
  "data": Object {
    "hash": "c22918aa52c9fdce78bf82451d8114e00b6d4375a8e5b1d1d96c6cf4755dbe1a",
    "packageName": "yargs",
    "version": "17.5.1",
  },
  "name": "npm:yargs",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes pnpm should map lock file data to external nodes 2`] = `
Array [
  Object {
    "source": "npm:yargs",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:escalade",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:get-caller-file",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:require-directory",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:y18n",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes pnpm should map successfully complex lock file 1`] = `
Object {
  "data": Object {
    "hash": "1e61b9db38c17534ce427d40ce84f9c03b4c439253f659e9a48513a10a52b465",
    "packageName": "nx",
    "version": "14.7.5",
  },
  "name": "npm:nx",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes pnpm should map successfully complex lock file 2`] = `
Array [
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/cli",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/tao",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@parcel/watcher",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chalk",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chokidar",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-cursor",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-spinners",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:dotenv",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:enquirer",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fast-glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:figures",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:flat",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fs-extra",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:ignore",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:js-yaml",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:jsonc-parser",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:minimatch",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:npm-run-path",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:open",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:semver",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tar-stream",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tmp",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tsconfig-paths",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tslib",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:v8-compile-cache",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes pnpm should map successfully complex lock file 3`] = `
Object {
  "data": Object {
    "hash": "6c5961e1a6ab89174727587860e480508f945407ce4c2cc7c9f1e46f181e6468",
    "packageName": "@phenomnomnominal/tsquery",
    "version": "4.1.1",
  },
  "name": "npm:@phenomnomnominal/tsquery",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes pnpm should map successfully complex lock file 4`] = `
Array [
  Object {
    "source": "npm:@phenomnomnominal/tsquery",
    "target": "npm:esquery",
    "type": "static",
  },
  Object {
    "source": "npm:@phenomnomnominal/tsquery",
    "target": "npm:typescript",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes yarn should map lock file data to external nodes 1`] = `
Object {
  "data": Object {
    "hash": "c22918aa52c9fdce78bf82451d8114e00b6d4375a8e5b1d1d96c6cf4755dbe1a",
    "packageName": "yargs",
    "version": "17.5.1",
  },
  "name": "npm:yargs",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes yarn should map lock file data to external nodes 2`] = `
Array [
  Object {
    "source": "npm:yargs",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:escalade",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:get-caller-file",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:require-directory",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:y18n",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes yarn should map successfully complex lock file 1`] = `
Object {
  "data": Object {
    "hash": "6562d56c136e9e359c74fb6843da8deb66890de8e61e49ee766ad9cee2be6878",
    "packageName": "nx",
    "version": "14.7.5",
  },
  "name": "npm:nx",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes yarn should map successfully complex lock file 2`] = `
Array [
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/cli",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/tao",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@parcel/watcher",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chalk",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chokidar",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-cursor",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-spinners",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:dotenv",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:enquirer",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fast-glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:figures",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:flat",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fs-extra",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:ignore",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:js-yaml",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:jsonc-parser",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:minimatch",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:npm-run-path",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:open",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:semver",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tar-stream",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tmp",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tsconfig-paths",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tslib",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:v8-compile-cache",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;
